src/main/java/org/autopotato/linkedinclonebackend/controllers/ResourceNotFoundException.java

```diff
@@ -0,0 +1,15 @@
+package org.autopotato.linkedinclonebackend.controllers;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+
+@EqualsAndHashCode(callSuper = true)
+@Value
+public class ResourceNotFoundException extends Exception {
```

Comment:
Usually, I would use the `RuntimeException` as the base class. But I can't say that there is a straightforward answer.
I guess we can try using caught exceptions and see how it goes.

For inspiration:
https://stackoverflow.com/questions/27578/when-to-choose-checked-and-unchecked-exceptions
https://stackoverflow.com/questions/499437/in-java-when-should-i-create-a-checked-exception-and-when-should-it-be-a-runti
https://stackoverflow.com/questions/6115896/understanding-checked-vs-unchecked-exceptions-in-java

Reply:
I agree, I changed it to `RuntimeException` ... I always forget that it exists 🙄